### ISSUE TITLE
SIGSEGV when using Apache 2.4 Event MPM + PHP 7.2 FPM + Snuffleupagus 0.4 on Ubuntu 18.04

### ISSUE THREAD

**Issue by cypherbits** [nbs-system/snuffleupagus#250]
Apache virtual host error log:

```
[Thu Sep 20 21:59:45.696280 2018] [proxy_fcgi:error] [pid 164:tid 139980730570496] [client 192.168.0.2:38226] AH01067: Failed to read FastCGI header
[Thu Sep 20 21:59:45.696398 2018] [proxy_fcgi:error] [pid 164:tid 139980730570496] (104)Connection reset by peer: [client 192.168.0.2:38226] AH01075: Error dispatching request to :
```

Php 7.2 FPM log:
```
[20-Sep-2018 21:59:45] WARNING: [pool www] child 97 exited on signal 11 (SIGSEGV - core dumped) after 73.844998 seconds from start
```

Disabling the Snuffleupagus extension fix the problem. v0.3.1 is working.

**Comment by jvoisin** [comment]
This is odd.
Can you please provide the coredump?

**Comment by cypherbits** [comment]
I can't right now, I'm using a Docker container on my working computer.
The bug seems to be in sp.sloppy_comparison.enable(), I disabled it and it works. I'm not using weird code, I'm just using SlimPhp and standard things for my project.
That means it should be related to the latests changes in there for in_array, array_search and array_keys.

**Comment by jvoisin** [comment]
It's a NULL-deref, I've found the issue, the reproducer (thanks to @kkadosh ) is on [a branch]( https://github.com/nbs-system/snuffleupagus/tree/fix_segfault_array_keys ).

I might fix it this week-end :)

**Comment by cypherbits** [comment]
Ok thanks.

**Comment by jvoisin** [comment]
Fixed in #252 thanks to @kkadosh and @xXx-caillou-xXx :)
